src/main/scala/exercise2/KarplusStrongAlgorithm.scala

```diff
@@ -0,0 +1,46 @@
+package exercise2
+
+import lib.StdAudio
+
+import scala.annotation.tailrec
+import scala.util.Random
+
+object KarplusStrongAlgorithm {
+  val energyDecayFactor = 0.996
+  val r = Random
+
+  def whiteNoiseHelper(volume: Double, resultQueue: QueueLike[Double], elementsLeft: Int): QueueLike[Double] = {
+    if (elementsLeft == 0) {
+      resultQueue
+    } else {
+      val value = (r.nextDouble() - 0.5) * volume
+      val updatedQueue = resultQueue.enqueue(value)
+      whiteNoiseHelper(volume, updatedQueue, elementsLeft-1)
+    }
+  }
+
+  def whiteNoise(frequency:Int=440, volume:Double = 1.0): QueueLike[Double] = {
+    whiteNoiseHelper(volume, Queue(), frequency)
```

Comment:
clean! Instead of using recursion, you could also achieve that by using generators and `foldLeft` with a starting value of an empty Queue.

```scala
(1 to frequency).map(_ => (Random.between(-0.5, 0.5)) * volume).foldLeft(.../*TODO*/)
```